exonum/src/encoding/fields.rs

```diff
@@ -258,6 +258,64 @@ impl<'a> Field<'a> for DateTime<Utc> {
     }
 }
 
+impl<'a> Field<'a> for Duration {
+    fn field_size() -> Offset {
+        (mem::size_of::<i64>() + mem::size_of::<i32>()) as Offset
+    }
+
+    unsafe fn read(buffer: &'a [u8], from: Offset, to: Offset) -> Self {
+        let secs = LittleEndian::read_i64(&buffer[from as usize..from as usize + 8]);
+        let nanos = LittleEndian::read_i32(&buffer[from as usize + 8..to as usize]);
+
+        // Assuming that buffer was checked and Duration object can be constructed.
+        Duration::seconds(secs) + Duration::nanoseconds(i64::from(nanos))
+    }
+
+    fn write(&self, buffer: &mut Vec<u8>, from: Offset, to: Offset) {
```

Comment:
Observation: This method seems to lead to a different presentation for negative durations than one used in `chrono` internally - it seems like `nanos` would be non-positive for these durations.